client/src/components/dashboard/Dashboard.js

```diff
@@ -1,23 +1,46 @@
 /* eslint-disable react/prop-types */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Link } from 'react-router-dom';
+import axios from 'axios';
 import auth from '../../auth/auth';
 
-const Dashboard = props => (
-  <>
-    <div>Dashboard</div>
-    <Link to="/">Home</Link>
-    <button
-      type="button"
-      onClick={() => {
-        auth.logout(() => {
-          props.history.push('/login');
-        });
-      }}
-    >
-Logout
-    </button>
-  </>
-);
+const Dashboard = (props) => {
+  const [userName, setUserName] = useState('');
+
+  useEffect(() => {
+    axios.get(process.env.REACT_APP_DASHBOARD_URL, {
```

Comment:
may be we need a networking layer that parsees tokens/data, and gives component the data it neds. A component fiddling with headers is not a great idea.